pkg/osquery/tables/firefox_preferences/table.go

```diff
@@ -0,0 +1,85 @@
+package firefox_preferences
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/launcher/pkg/dataflatten"
+	"github.com/kolide/launcher/pkg/osquery/tables/dataflattentable"
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/osquery/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+type Table struct {
+	name   string
+	logger log.Logger
+}
+
+const tableName = "kolide_firefox_preferences"
+
+var re = regexp.MustCompile(`^user_pref\((.*)\)`)
+
+func TablePlugin(logger log.Logger) *table.Plugin {
+	columns := dataflattentable.Columns(
+		table.TextColumn("path"),
+	)
+
+	t := &Table{
+		name:   tableName,
+		logger: logger,
+	}
+
+	return table.NewPlugin(t.name, columns, t.generate)
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	return generateData(queryContext, t.logger)
+}
+
+func generateData(queryContext table.QueryContext, logger log.Logger) ([]map[string]string, error) {
+	paths := tablehelpers.GetConstraints(queryContext, "path")
+
+	if len(paths) == 0 {
+		return nil, errors.Errorf("The %s table requires that you specify a constraint for path", tableName)
+	}
+
+	rowData := map[string]string{"path": paths[0]}
```

Comment:
Just noticed this `paths[0]` leftover from before I allowed multiple. I'll fix this up.